test/regtest.py

```diff
@@ -105,10 +114,24 @@ def process_test(cmd, test, memory, verifier, expect, log_file):
   p = subprocess.Popen(cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
   out, err  = p.communicate()
   elapsed = time.time() - t0
+  status = 0
+
+  bplfile = cmd[cmd.index('-bpl')+1]
+  with open(os.devnull, 'w') as devnull:
+    for f in checkbpl:
+      with open(bplfile) as bpl:
+        checker = subprocess.Popen(shlex.split(f), stdin=bpl, stdout=devnull, stderr=devnull)
```

Comment:
Because `grep` is just kind of a degenerate case of checking the generated Boogie file. I think we want to allow for more flexibility, hence the checking predicate can be an arbitrary command. E.g., you may want to type-check the generated Boogie code, which can be done by wrapping up `boogie /noVerify /doModSetAnalysis`. E.g., you may want to run some simple analysis on the generated Boogie code, which might be done by `bam --my-analysis`. This implementation is the simplest way of being relatively future-proof.